code/modules/admin/verbs/debug.dm

```diff
@@ -889,3 +889,59 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
 		return
 	if(alert(usr, "Are you absolutely sure you want to reload the configuration from the default path on the disk, wiping any in-round modificatoins?", "Really reset?", "No", "Yes") == "Yes")
 		config.admin_reload()
+
+/// A debug verb to check the sources of currently running timers
+/client/proc/check_timer_sources()
+	set category = "Debug"
+	set name = "Check Timer Sources"
+	set desc = "Checks the sources of the running timers"
+	if (!check_rights(R_DEBUG))
+		return
+
+	var/bucket_list_output = generate_timer_source_output(SStimer.bucket_list)
+	var/second_queue = generate_timer_source_output(SStimer.second_queue)
```

Comment:
bucket_list is a fixed sized list of linked lists.

Meaning each timer in the bucket list is only the HEAD of a linked list. `generate_timer_source_output` here is only looking at the head timer and not any other item in the linked list.

next/prev vars will be null on secondqueue, so you could safely add a linked list transversal to generate_timer_source_output to fix this.

see reset_buckets for code designed to loop through all timers https://github.com/tgstation/tgstation/blob/master/code/controllers/subsystem/timer.dm#L236